app/src/main/java/org/schabi/newpipe/util/ShareUtils.java

```diff
@@ -149,8 +158,9 @@ private static void openInDefaultApp(final Context context, final Intent viewInt
      * @return the package name of the default app, or the app chooser if there's no default
      */
     private static String getDefaultAppPackageName(final Context context, final Intent intent) {
-        return context.getPackageManager().resolveActivity(intent,
-                PackageManager.MATCH_DEFAULT_ONLY).activityInfo.packageName;
+        final ResolveInfo resolveInfo = context.getPackageManager().resolveActivity(intent,
+                PackageManager.MATCH_DEFAULT_ONLY);
+        return resolveInfo.activityInfo.packageName;
```

Comment:
This will still throw a null pointer exception

Reply:
What do you think about this?
```
    private static String getDefaultBrowserPackageName(final Context context) {
        final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("http://"))
                .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        String packageName = "";
        try {
            packageName = context.getPackageManager().resolveActivity(intent,
                    PackageManager.MATCH_DEFAULT_ONLY).activityInfo.packageName;
        } catch (final NullPointerException e) {
            return null;
        }
        return packageName;
    }
```
```
    private static String getDefaultAppPackageName(final Context context, final Intent intent) {
        String packageName = "";
        try {
            packageName = context.getPackageManager().resolveActivity(intent,
                    PackageManager.MATCH_DEFAULT_ONLY).activityInfo.packageName;
        } catch (final NullPointerException e) {
            return null;
        }
        return packageName;
    }
```

Reply:
```
    private static String getDefaultAppPackageName(final Context context, final Intent intent) {
        try {
            return context.getPackageManager().resolveActivity(intent,
                    PackageManager.MATCH_DEFAULT_ONLY).activityInfo.packageName;
        } catch (final NullPointerException e) {
            return null;
        }
    }
```